strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/test/java/org/carlspring/strongbox/artifact/generator/NpmPackageGenerator.java

```diff
@@ -66,6 +67,15 @@ public NpmPackageGenerator in(Path path)
         return this;
     }
 
+    @Override
+    public Path generateArtifact(URI uri,
+                                 int size)
+            throws IOException
+    {
+        this.in(Paths.get(uri));
+        return(this.buildPublishJson());
```

Comment:
you should return `getPackagePath()`. Also this is not required to use `this` everywhere.  :)